Webpack angular2.0

rvanmarkus opened the pull request:
@RobQuistNL can you test this branch on your machine? New frontend with webpack setup, see  seedproject (https://github.com/AngularClass/angular2-webpack-starter)

Any suggestions for somehow fixing the massive amounts of commits from the pull of angularClass repo? Or other comments? ;) Otherwise please merge into master! 👍

PatrickJS committed:
refactor: vendor/polyfills as browser

PatrickJS committed:
refactor(platform): change file format

PatrickJS committed:
docs: hmr comment

PatrickJS committed:
refactor(home): markdown

PatrickJS committed:
chore(spec-bundle): use direct polyfills

PatrickJS committed:
docs(readme): typings install

PatrickJS committed:
docs(custom-typings): typings install

PatrickJS committed:
refactor: remove provide helper

PatrickJS committed:
chore: update deps

leonardlin committed:
fix(AppState): bool false value not returned

this resolves the issue #673, that boolean properties with value 'false' are not returned.

shlomiassaf committed:
feat(configurable-head-tags): allow configuring head tags in a js module (#679)

* feat(configurable-head-tags): allow configuring head tags in a js module

* fix: lint code

PatrickJS committed:
chore: update copy-webpack-plugin

PatrickJS committed:
refactor(index.html): best practices for head

shlomiassaf committed:
feat(github-deploy): deploy to github pages (#645)

MehulATL committed:
chore: update deps (#698)

PatrickJS committed:
feat(environment): provide workaround prodmode

PatrickJS committed:
fix(environment): correct compiler config

PatrickJS committed:
chore: include forms

PatrickJS committed:
chore: remove legacy polyfills

PatrickJS committed:
chore(webpack.prod): replace angular2-hmr with noop

PatrickJS committed:
style(webpack.prod): space

PatrickJS committed:
chore(webpack): use current version of CopyWebpackPlugin

PatrickJS committed:
refactor(webpack): use tostring for css

PatrickJS committed:
chore(vendor): include hmr

PatrickJS committed:
fix(webpack): correct import

PatrickJS committed:
chore(webpack.test): correct css loaders

PatrickJS committed:
feat: use new @angular/form api

PatrickJS committed:
chore: update typings 1.3.0

PatrickJS committed:
chore: update awesome-typescript-loader 1.1.1

alujs committed:
docs(readme): use markdown correctly

PatrickJS committed:
chore: remove root dir for profiling

PatrickJS committed:
chore(tslint): allow null

izifortune committed:
core: globbing problems on tslint, on osx wouldn't correctly glob (#699)

PatrickJS committed:
chore(polyfills): include ie-shim

PatrickJS committed:
chore: update typedoc 0.4.3

PatrickJS committed:
chore: update ts-node 0.9.1

PatrickJS committed:
chore: update html-webpack-plugin 2.21.0

PatrickJS committed:
chore(polyfills): remove core-js Promise

PatrickJS committed:
docs(readme): windows only install

PatrickJS committed:
chore: revert awesome-typescript-loader 0.17.0

PatrickJS committed:
chore(typings): remove old types

PatrickJS committed:
chore: checkNodeImport in helpers

PatrickJS committed:
chore: update angular2-hmr 0.8.0

PatrickJS committed:
chore: include @angular/forms vendor

PatrickJS committed:
Update README.md

PatrickJS committed:
feat(browser-directives): include reactive form directives

PatrickJS committed:
chore: update to latest router

PatrickJS committed:
chore: include @angularclass/webpack-toolkit

PatrickJS committed:
chore: remove router-active

PatrickJS committed:
refactor: attend .template .style

PatrickJS committed:
chore: update to latest router

PatrickJS committed:
chore: create routes file with async routes

PatrickJS committed:
refactor(routes): use RouterConfig

PatrickJS committed:
style(app.routes): use full loader name

PatrickJS committed:
feat: use angular2-template-loader

PatrickJS committed:
chore: update file-loader

PatrickJS committed:
feat: NoContent 404

PatrickJS committed:
refactor(app): include footer

PatrickJS committed:
Update README.md

PatrickJS committed:
docs(app.routes): comment asyncRoutes

PatrickJS committed:
feat(environment): enableDebugTools

PatrickJS committed:
refactor(environment): typo

kulshekhar committed:
chore: Remove a duplicate dev dependency (#715)

PatrickJS committed:
Update webpack.prod.js

PatrickJS committed:
chore: update Angular 2.0.0-rc.3

PatrickJS committed:
feat: @angularclass/request-idle-callback

PatrickJS committed:
core: remove legacy 3rd party module backport

PatrickJS committed:
chore: correct deps vs devDeps

PatrickJS committed:
Update typings.json

PatrickJS committed:
chore: update @angularclass/request-idle-callback

PatrickJS committed:
chore: update @angularclass/request-idle-callback

Commit:
chore(typings): update typings (#719)

* chore(typings): update typings

* chore(typings): update node typings

PatrickJS committed:
Update README.md

PatrickJS committed:
docs(app.routes): comments about routes

PatrickJS committed:
chore: update @angularclass/webpack-toolkit

PatrickJS committed:
chore: update @angularclass/request-idle-callback

PatrickJS committed:
docs(home): ngModel with name

PatrickJS committed:
chore: remove PhantomJS

PatrickJS committed:
chore: update router to alpha.8

PatrickJS committed:
chore: update @angularclass/webpack-toolkit

PatrickJS committed:
chore: include async folder

PatrickJS committed:
chore: update typings

PatrickJS committed:
chore: include +details

PatrickJS committed:
refactor(+detail): move console.log to index

PatrickJS committed:
feat: include ngOnInit and constructor for +detail

PatrickJS committed:
chore: use awesome-typescript-loader 1.1.1

PatrickJS committed:
docs(main.browser): comment about vendor providers

PatrickJS committed:
chore: update typedoc

PatrickJS committed:
refactor: correctly use async component

PatrickJS committed:
chore: provide var lib for dev

PatrickJS committed:
chore: retain ng.probe ng.coreTokens

PatrickJS committed:
fix: initial load of async child routes

PatrickJS committed:
chore: update angular 2.0.0-rc.4

PatrickJS committed:
feat: update @angular/router 3.0.0-beta.1

PatrickJS committed:
feat: example of resolved data route

PatrickJS committed:
Update app.routes.ts

PatrickJS committed:
chore: update router beta.2

PatrickJS committed:
test: fix tests

PatrickJS committed:
chore(typings): update WebpackRequire

PatrickJS committed:
chore(package.json): typings-install

acdcjunior committed:
docs(readme): fix typings install command (#747)

...since there's no `typings-install` script @ package.json.

PatrickJS committed:
docs(readme): typings-install

PatrickJS committed:
types: inline WebpackRequireEnsureCallback

PatrickJS committed:
chore: update custom-typings

PatrickJS committed:
chore: moduleResolution: node

PatrickJS committed:
Update custom-typings.d.ts

PatrickJS committed:
feat: update typescript, remove typings

PatrickJS committed:
chore: fix inline mode

PatrickJS committed:
chore(package.json): ts 2.0.0-beta

PatrickJS committed:
chore: update @types

PatrickJS committed:
Update README.md

alerodrod committed:
docs(readme): fix humans.txt reference (#806)

PatrickJS committed:
Update tslint.json

PatrickJS committed:
Update tslint.json

PatrickJS committed:
feat(Angular2): Free Angular 2 Fundamentals Course (#818)

<img width="438" alt="Angular 2 Fundamentals" src="https://cloud.githubusercontent.com/assets/1016365/17200649/085798c6-543c-11e6-8ad0-2484f0641624.png">

# Coming Soon

PatrickJS committed:
feat(Angular2): Free Angular 2 Fundamentals Course

PatrickJS committed:
feat: switch to @angularclass/conventions-loader

PatrickJS committed:
feat: use @angularclass/conventions-loader

PatrickJS committed:
Update package.json

PatrickJS committed:
Update webpack.common.js

PatrickJS committed:
Update tsconfig.json

PatrickJS committed:
feat: process true

theRemix committed:
chore: Fixes initial e2e tests (#837)

* fixes initial e2e tests

resolves #836

* Test/fix travis xvfb (#2)

* add beforescript for travisci to resolve chrome issue with xvfb

* travis.yml before_script typo

* travisci use chromium-browser

* add Chrome_travis_ci browser to karma conf

martaver committed:
feat: Added use of file-loader for jpg/png/gif files

Added use of file-loader for jpg/png/gif files. This adds support for using background url() statements in CSS files, as per AngularClass#759

Pyro979 committed:
feat: Update webpack.dev.js to take respect optional HOST/PORT params (#827)

Fixes issue:
https://github.com/AngularClass/angular2-webpack-starter/issues/821
And partially:
https://github.com/AngularClass/angular2-webpack-starter/issues/186


Now to run on a VM you can just run
PORT=8081 HOST="0.0.0.0" npm run server:dev:hmr
or if you're using "forever"
PORT=8081 HOST="0.0.0.0" forever start -v --minUptime 1000 --spinSleepTime 1000 -c "npm run server:dev:hmr" ./

Commit:
chore: add comments (#825)

cebor committed:
use typescript declarations for es2015 (#792)

PatrickJS committed:
Update tsconfig.json

PatrickJS committed:
chore: core-js

jiverson committed:
chore: update deprecated test functions (#842)

adams committed:
update utm

PatrickJS committed:
Feature/ng2 rc5 (#875)

* Version bump to rc.5

* Migrates to RC 5: adds AppModule, re-organizes providers

* Migrates to RC5: re-organize routing for async/lazy routes. still an issue though

PatrickJS committed:
fix: bootstrap return componentRef

PatrickJS committed:
fix: async, hmr, remove platform

PatrickJS committed:
Update package.json

PatrickJS committed:
Update app.module.ts

PatrickJS committed:
fix: hmr

PatrickJS committed:
feat: update webpack2

PatrickJS committed:
Update vendor.browser.ts

PatrickJS committed:
Update polyfills.browser.ts

PatrickJS committed:
Update README.md

isaacplmann committed:
Include string-replace-loader as devDependency (#899)

PatrickJS committed:
Update app.service.ts

PatrickJS committed:
fix(hmr): tick on change

likerRr committed:
docs: update comment (#907)

StefH committed:
fix: TestComponentBuilder  (#904)

StefH committed:
fix: #901 (#905)

greenkeeperio-bot committed:
chore(package): update css-loader to version 0.24.0 (#895)

https://greenkeeper.io/

Commit:
fix: tslint issues (#870)

greenkeeperio-bot committed:
chore(package): update angular2-template-loader to version 0.5.0 (#894)

https://greenkeeper.io/

katallaxie committed:
Add support for webpack-dashboard and NamedModulePlugin

* Using webpack-dashboard in dev mode
* Adding NamedModulePlugin in dev mode

katallaxie committed:
Add support for webpack-dashboard and NamedModulePlugin

* Using webpack-dashboard in dev mode
* Adding NamedModulePlugin in dev mode

katallaxie committed:
Fixing TravisCI with npm >= 3

katallaxie committed:
Fixing TravisCI with npm >= 3

katallaxie committed:
Adding .vscode for Chrome Debugger

katallaxie committed:
Adding .vscode for Chrome Debugger

katallaxie committed:
ESLint for all config/**/*.js

katallaxie committed:
ESLint for all config/**/*.js

katallaxie committed:
~ updating deps in dev

katallaxie committed:
~ updating deps in dev

katallaxie committed:
~ fixing eslint and tslint

katallaxie committed:
~ fixing eslint and tslint

katallaxie committed:
~ updating TSLint to @next

katallaxie committed:
~ updating TSLint to @next

katallaxie committed:
~ removing conflicting @types/core-js definitions

katallaxie committed:
~ removing conflicting @types/core-js definitions

0cv committed:
fix: Update webpack.prod.js (#920)

Temporary Fix for RC5. See https://github.com/angular/angular/issues/10618

greenkeeperio-bot committed:
chore(package): update es6-promise-loader to version 1.0.2 (#893)

https://greenkeeper.io/

PatrickJS committed:
chore: remove es6-promise-loader for System.import

zhenwenc committed:
chore(package): move angular2-template-loader to devDependencies (#909)

greenkeeperio-bot committed:
chore(package): update tslint to version 3.15.1 (#883)

https://greenkeeper.io/

katallaxie committed:
~ remove eslint and webpack-dashboard

katallaxie committed:
~ remove eslint and webpack-dashboard

katallaxie committed:
~ reverting minor failure

katallaxie committed:
~ reverting minor failure

katallaxie committed:
~ updating webpack.prod.js

katallaxie committed:
~ updating webpack.prod.js

katallaxie committed:
Merge remote-tracking branch 'origin/master'

PatrickJS committed:
Merge: Adding Named Modules, VS Code, Fixes

Adding Named Modules, VS Code, Fixes

Commit:
remove unnecessary blank line

katallaxie committed:
feat: ~ adding .nvmrc and respective docs

katallaxie committed:
Merge branch 'master' of https://github.com/AngularClass/angular2-webpack-starter

katallaxie committed:
~ fixing typescript@2.0.0-rc changes and VS Code

likerRr committed:
Replace obsolete setBaseTestProviders with TestBed.initTestEnvironment

katallaxie committed:
Merge pull request #939 from likerRr/update_test

Replace obsolete setBaseTestProviders with TestBed.initTestEnvironment

StefH committed:
Replace obsolete addProviders by TestBed.configureTestingModule (#929)

Replace obsolete addProviders by TestBed.configureTestingModule

katallaxie committed:
~ doing some linting in specs

orizens committed:
docs(readme): added @types example (#942)

* added @types example

* updated conditions on when to add @types

katallaxie committed:
Updating to angular2-*@2.0.0-rc6 (deps, specs ...) (#944)

* ~ adding angular-*@2.0.0-rc6
* ~ fixing .specs for angular-*@2.0.0-rc6
* ~ removing comment from x-large spec
* ~ remove old module from +detail and adding again CommonModule
* ~ disabling DedupePlugin (https://github.com/angular/angular-cli/issues/1587)

katallaxie committed:
~ updating karma@1.2.0

katallaxie committed:
~ updating webpack-merge@0.14.1 and webpack-dev-server@2.1.0-beta.2

katallaxie committed:
~ updating core-js@2.4.1, codelyzer@0.0.28, karama-coverage@1.1.1

katallaxie committed:
~ updating shrinkwrap

PatrickJS committed:
Update package.json

aaalsubaie committed:
refactor(helper): checkNodeImport is never used! (#844)

katallaxie committed:
~ adding support for building a Docker from build:prod (#947)

* ~ adding `Dockerfile`
* ~ adding testing to `.travis.yml`
* ~ doing some `foo`

greenkeeperio-bot committed:
chore(package): update css-loader to version 0.25.0 (#963)

https://greenkeeper.io/

kondi committed:
feat: Add missing type information to compile in case of `noImplicitAny` (#928)

PatrickJS committed:
chore(npm-shrinkwrap.json): too confusing when installing new modules

too confusing when installing new modules

PatrickJS committed:
chore(package.json): use patched string-replace-loader

rzubov committed:
fix(helpers): isWebpackDevServer() fix for windows, issue #757 (#966)

Fix #757

PatrickJS committed:
fix(webpack): check if default for es5 module

PatrickJS committed:
refactor(webpack): better es5 check for default

PatrickJS committed:
chore: update webpack and hmr-loader

PatrickJS committed:
feat: save dom input values

PatrickJS committed:
chore: update hmr

PatrickJS committed:
feat(hmr): createInputTransfer

PatrickJS committed:
docs(faq): windows/node path problem

greenkeeperio-bot committed:
chore(package): update typedoc to version 0.4.5 (#868)

https://greenkeeper.io/

PatrickJS committed:
refactor(about): update with System.import

avoyage committed:
refactor(main.browser.ts): remove unused import (#976)

PatrickJS committed:
feat: restoreInputValues for async routes

PatrickJS committed:
fix: angular2 perf problem

closes #979

PatrickJS committed:
chore: update hmr-loader

Commit:
chore: Update angular to version 2.0.0 🚀 (#1009)

* Update angular to version 2.0.0 :rocket:

* Update angular/router to version 3.0.0

Simon-Briggs committed:
fix: Angular2 dependency warning (#998)

* fix: Angular2 RC7 dependency warning

* ~ minor syntactic change

LGTM

* Update webpack.common.js

rvanmarkus committed:
empty working dir for clean install

rvanmarkus committed:
Merge branch 'master' of https://github.com/angularclass/angular2-webpack-starter into webpack-angular2.0

rvanmarkus committed:
- merged main application logic of the client in new angular2.0 stable project, without material design
- syntax fixes for 2.0 release

rvanmarkus committed:
- modules navlink

rvanmarkus committed:
- scss webpack config
- clock header
-

RobQuistNL committed:
Some optical changes to frontend

rvanmarkus committed:
rx.Observable => Observable

rvanmarkus committed:
PS-39; haal API_URL en API_PORT uit environment variables

rvanmarkus committed:
production environment variables for API_URL and API_PORT

rvanmarkus committed:
- added md-icon module for dynamic loading and rendering SVG / font icons and  lateron handling different icon registries for modules
- added google material icons as default iconset

rvanmarkus committed:
- added md-icon module for dynamic loading and rendering SVG / font icons and  lateron handling different icon registries for modules
- added google material icons as default iconset

rvanmarkus committed:
directly show clock value before wait till first interval tick

rvanmarkus committed:
directly show clock value before wait till first interval tick

RobQuistNL commented:
I'm suggesting; 

```
rm .git -rf
git init
git remote add origin git@github.com:PiStation/pistation-client.git
git add .
git commit -m "Initial commit (webpack-agular2 + small changes)"
git push origin master -f
```

Then we can close this pr

sshev committed:
feat: show test coverage for original TS code (#1014)

* add karma-remap-coverage reporter to map coverage to *.ts

dangerdespain committed:
Webpack 2.1.0-beta.22 (#1018)

* Webpack 2.1.0-beta.22
* Braking Changes in https://github.com/webpack/webpack/releases/tag/v2.1.0-beta.23
* further work needed, just to fix Travis

* ~ updating @types

* ~ pinning @types/selenium-webdriver

greenkeeperio-bot committed:
chore(package): update typescript to version 2.0.3 (#1040)

https://greenkeeper.io/

mmrath committed:
Update string-replace-loader to 1.0.5 (#1032)

This removes the following warning about webpack dependency.

npm WARN string-replace-loader@1.0.3 requires a peer of webpack@1.x.x || 2.x.x but none was installed.

ashleahhill committed:
Set karma-webpack to only log errors (#1005)

Set wepack to only output errors during testing

akomm committed:
chore(webpack) (#987)

* moved resolve.root to resolve.modules

katallaxie committed:
chore(dev) PhantomJS (#946)

* ~ **do not** love PhantomJS, but as to have unit test running in the `console` und in most CIs

harunurhan committed:
Remove @types/corejs (#1049)

jokecamp committed:
Update tsconfig.json paths property to object (#1073)

In the tsconfig.json the property `compilerOptions.paths` should be an object and not an array.

See https://www.typescriptlang.org/docs/handbook/compiler-options.html

hongbo-miao committed:
docs(readme): update the way to use sass/scss (#1074)

This is the working way for the version now

mzellho committed:
removed density property from manifest.json (#1069)

nchutchind committed:
Copy humans.txt and robots.txt to dist/ (#1076)

Fixes issue #1070

The standard for humans.txt and robots.txt says that they should be available from the root of the application. This change will make sure that they get copied from the assets folder to the dist/ root instead of to dist/assets.

dherges committed:
Remove unsed dependencies: webpack-toolkit, request-idle-callback (#1080)

olegskl committed:
refactor(app): fix typo in InteralStateType (#1083)

Renamed `InteralStateType` to `InternalStateType`.

dherges committed:
Embraces ng2 styleguide *.component.{css|html|ts} (#1084)

katallaxie committed:
chore(npm): ~ removing shrinkwrap

lostdalek committed:
chore(package) typo (#1099)

greenkeeperio-bot committed:
chore(package): update ts-helpers to version 1.1.2 (#1102)

https://greenkeeper.io/

dherges committed:
Adjustments to the Angular 2 Style Guide (#1091)

* Component class names follow the style guide
* Filenames of services, components, and directives follow the style guide

Commit:
Merge branch 'master' of github.com:AngularClass/angular2-webpack-starter into webpack-angular2.0

Conflicts:
	package.json
	src/app/app.component.css
	src/app/app.component.ts
	src/app/app.module.ts
	src/app/app.routes.ts

Commit:
Merge branch 'master' of github.com:AngularClass/angular2-webpack-starter into webpack-angular2.0

Conflicts:
	package.json
	src/app/app.component.css
	src/app/app.component.ts
	src/app/app.module.ts
	src/app/app.routes.ts

rvanmarkus commented:
Still needs to happen! @RyuuMasato kan jij wellicht kijken, met de comments van Rob in gedachte?

RobQuistNL commented:
If we do that we have a clean slate to start with :)
